salt/states/slack.py

```diff
@@ -39,12 +39,7 @@ def __virtual__():
     return 'slack' if 'slack.post_message' in __salt__ else False
 
 
-def post_message(name,
-                 channel,
-                 from_name,
-                 message,
-                 api_key=None,
-                 icon=None):
+def post_message(name, **kwargs):
```

Comment:
Changing the signature to `**kwargs` is an immediate red flag.
Let's not.
Or better, why do we need it?
To support old and new signature?
Add old and new and deprecate old, etc...

Reply:
This is weird. The modified function is doing some validation then dispatching calls to slack_notify.py's `call_hook` and `post_message` (both of which do their own validation, by the way).

This function is called `post_message`, so why would it ever call `call_hook`?

I can see adding a generic function called "send_message" if it made sense, but it doesn't:

The only arguments in common are `message` and `channel`.

So long as the caller needs to keep track of the method of dispatch and the several unique arguments, it is better to have 2 functions that are named differently.

-----------

This could be 2 functions, `post_message` and `call_hook` which just pass through to the the `slack_notify` module. 

Why would someone use this instead of `module.run`? Is that less common?